File: Proyecto-PracticasProfesionales-main-corregido/backend/interfaces/api/schemas/bitacora_auditoria.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class BitacoraAuditoriaCreate(BaseModel):
    id_usuario: int | None = None
    accion: str
    modulo: str
    descripcion: str | None = None
    entidad: str | None = None
    id_entidad: int | None = None
    ip: str | None = None
    user_agent: str | None = None


class BitacoraAuditoriaResponse(BaseModel):
    id_bitacora: int
    id_usuario: int | None = None
    usuario_correo: str | None = None
    accion: str
    modulo: str
    descripcion: str | None = None
    entidad: str | None = None
    id_entidad: int | None = None
    fecha: datetime
    ip: str | None = None
    user_agent: str | None = None

    model_config = ConfigDict(from_attributes=True)

File: Proyecto-PracticasProfesionales-main-corregido/src/domain/admin/BitacoraAuditoria.ts
export interface BitacoraAuditoriaItem {
  id_bitacora: number;
  id_usuario: number | null;
  usuario_correo: string | null;
  accion: string;
  modulo: string;
  fecha: string | null;
  descripcion: string | null;
  entidad: string | null;
  id_entidad: number | null;
  ip: string | null;
  user_agent: string | null;
}

export interface BitacoraAuditoriaFiltros {
  fecha_inicio?: string;
  fecha_fin?: string;
  modulo?: string;
  accion?: string;
  entidad?: string;
  usuario?: string;
  pagina?: number;
  limite?: number;
}

export interface BitacoraAuditoriaResponse {
  items: BitacoraAuditoriaItem[];
  total: number;
  pagina: number;
  limite: number;
  total_paginas: number;
  modulos: string[];
  acciones: string[];
  entidades: string[];
}
